### ISSUE TITLE
Assign shortcuts to extension actions

### ISSUE THREAD

**Issue by uliska** [wbsoft/frescobaldi#1136]
If I see correctly the most pressing open question after merging #1129 (and in my view the only thing that I wouldn't want to be open after a 3.1 release) is the ability to assign keyboard shortcuts to the actions of an extension.

I haven't implemented this because I was somewhat afraid of it and I haven't yet really looked into how the shortcuts are managed. In a way my code actively prevents an extension's ActionCollection to be handed over to the global shortcut management. So maybe it is very easy to have Frescobaldi also manage the actions from extensions.

However, one thing that should be properly considered is how to handle shortcuts that are assigned to extensions that are deactivated (on purpose or due to errors) or even uninstalled.

**Comment by PeterBjuhr** [comment]
Could it be compared to assigning shortcuts to snippets the users have created themselves?

**Comment by uliska** [comment]
That sounds like a promising idea, at least the issue with removed snippets should be very comparable. I'll try looking into this ASAP.

**Comment by ramondor** [comment]
Hi Guys, I think the extensions functionality is great. I'm planning to add an extensive set of actions for navigating and editing for composers share with the community. However, without the ability to assign keyboard shortcuts to the actions, the advantage of calling them on the fly while typing lilypond code is lost. For the time being, is there any conceivable workaround to this? or any quick fix at least for now to hack shortcuts to work? thanks for your great work!

**Comment by ramondor** [comment]
as a really bad workaround, would coding a python snippet to call an extension action work?

**Comment by uliska** [comment]
Vety short abswer: that should work. Extensions end up as regular Python modules in Frescobaldi's Python path
-- 
Diese Nachricht wurde von meinem Android-Gerät mit K-9 Mail gesendet.

**Comment by ramondor** [comment]
tnx! I guess you refer to using the snippets. Is this also the best option for now?

**Comment by uliska** [comment]
Yes, I do. 
Since I don't have time to look into it (although this is one of the things I know need to be added) it is probably the most straightforward workaround for now.
-- 
Diese Nachricht wurde von meinem Android-Gerät mit K-9 Mail gesendet.

**Comment by ramondor** [comment]
thanks @uliska ! I would appreciate some pointers before I start coding something which does not fit in your design:
1. the extension should intercept keyboard input and manipulate the document based on that input
2. it should stay active as long as the user wishes to make use of that input interception, yet not block the actions carried out. so if, as a result of user input some lilypond code is rewritten, the rewrite should take place while the extension is still running, so that the document content is updated and the user can carry on using the extension without reloading it. 

in short, what is needed from the extension is not a one shot action, but a continuous non-blocking keyboard interception layer (with quite a lot of logic in the middle). 
is this possible to achieve that using only the mainwindow() etc. that i get from extensions.Extension, or do i need to instantiate an additional MainWindow(QWidget)?

how would you recommend approaching this? 
thanks, Ramon
